cors.go

```diff
@@ -1 +1,126 @@
+// Copyright 2021 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package cors is a middleware that generates CORS headers for Flamego.
 package cors
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/flamego/flamego"
+)
+
+const anyDomain = "!*"
+
+// Options contains options for the cors.CORS middleware.
+type Options struct {
+	// Scheme may be http or https as accepted schemes or the '*' wildcard to accept any scheme. (default: "http")
+	Scheme string
+	// AllowDomain is a comma separated list of domains that are allowed to run CORS requests
+	// Special values are the  a single '*' wildcard that will allow any domain to send requests without
+	// credentials and the special '!*' wildcard which will reply with requesting domain in the 'access-control-allow-origin'
+	// header and hence allow requests from any domain *with* credentials. (default '*')
+	AllowDomain []string
+	// AllowSubdomain allowed subdomains of domains to run CORS requests. (default false)
+	AllowSubdomain bool
+	// Methods may be a comma separated list of HTTP-methods to be accepted. (default GET, POST, OPTIONS)
+	Methods []string
+	// MaxAgeSeconds may be the duration in secs for which the response is cached. (default 600)
+	MaxAgeSeconds int
+	// AllowCredentials set to false rejects any request with credentials. (default false)
+	AllowCredentials bool
+}
+
+func prepareOptions(options []Options) Options {
+	var opt Options
+	if len(options) > 0 {
+		opt = options[0]
+	}
+
+	if len(opt.Scheme) == 0 {
```

Comment:
```suggestion
	if opt.Scheme == "" {
```

The idiomatic Go is to compare with empty string when checking if the string is empty.